interfaces/apparmor/spec_test.go

```diff
@@ -144,26 +144,141 @@ apps:
-			"# Layout path: /usr\n/usr{,/**} mrwklix,",
+			"# Layout path: /usr/foo\n/usr/foo{,/**} mrwklix,",
+			"# Layout path: /var/cache/mylink\n# (no extra permissions required for symlink)",
+			"# Layout path: /var/tmp\n/var/tmp{,/**} mrwklix,",
 		},
 	})
+
+	profile0 := `  # Layout /etc/foo.conf: bind-file $SNAP/foo.conf
+  mount options=(bind, rw) /snap/vanguard/42/foo.conf -> /etc/foo.conf,
+  umount /etc/foo.conf,
+  # Writable mimic /etc
+  mount options=(rbind, rw) /etc/ -> /tmp/.snap/etc/,
+  mount fstype=tmpfs options=(rw) tmpfs -> /etc/,
+  mount options=(rbind, rw) /tmp/.snap/etc/** -> /etc/**,
+  mount options=(bind, rw) /tmp/.snap/etc/* -> /etc/*,
+  umount /tmp/.snap/etc/,
+  umount /etc{,/**},
+  /etc/** rw,
+  /etc/ rw,
+  /tmp/.snap/etc/** rw,
+  /tmp/.snap/etc/ rw,
+  /tmp/.snap/ rw,
+  /tmp/ rw,
+  # Writable mimic /snap/vanguard/42
+  mount options=(rbind, rw) /snap/vanguard/42/ -> /tmp/.snap/snap/vanguard/42/,
+  mount fstype=tmpfs options=(rw) tmpfs -> /snap/vanguard/42/,
+  mount options=(rbind, rw) /tmp/.snap/snap/vanguard/42/** -> /snap/vanguard/42/**,
+  mount options=(bind, rw) /tmp/.snap/snap/vanguard/42/* -> /snap/vanguard/42/*,
+  umount /tmp/.snap/snap/vanguard/42/,
+  umount /snap/vanguard/42{,/**},
+  /snap/vanguard/42/** rw,
+  /snap/vanguard/42/ rw,
+  /snap/vanguard/ rw,
+  /snap/ rw,
+  /tmp/.snap/snap/vanguard/42/** rw,
+  /tmp/.snap/snap/vanguard/42/ rw,
+  /tmp/.snap/snap/vanguard/ rw,
+  /tmp/.snap/snap/ rw,
+  /tmp/.snap/ rw,
+  /tmp/ rw,
+`
+	profile1 := `  # Layout /usr/foo: bind $SNAP/usr/foo
+  mount options=(rbind, rw) /snap/vanguard/42/usr/foo/ -> /usr/foo/,
+  umount /usr/foo/,
+  # Writable mimic /usr
+  mount options=(rbind, rw) /usr/ -> /tmp/.snap/usr/,
+  mount fstype=tmpfs options=(rw) tmpfs -> /usr/,
+  mount options=(rbind, rw) /tmp/.snap/usr/** -> /usr/**,
+  mount options=(bind, rw) /tmp/.snap/usr/* -> /usr/*,
+  umount /tmp/.snap/usr/,
+  umount /usr{,/**},
+  /usr/** rw,
+  /usr/ rw,
+  /tmp/.snap/usr/** rw,
+  /tmp/.snap/usr/ rw,
+  /tmp/.snap/ rw,
+  /tmp/ rw,
+  # Writable mimic /snap/vanguard/42/usr
+  mount options=(rbind, rw) /snap/vanguard/42/usr/ -> /tmp/.snap/snap/vanguard/42/usr/,
+  mount fstype=tmpfs options=(rw) tmpfs -> /snap/vanguard/42/usr/,
+  mount options=(rbind, rw) /tmp/.snap/snap/vanguard/42/usr/** -> /snap/vanguard/42/usr/**,
+  mount options=(bind, rw) /tmp/.snap/snap/vanguard/42/usr/* -> /snap/vanguard/42/usr/*,
+  umount /tmp/.snap/snap/vanguard/42/usr/,
+  umount /snap/vanguard/42/usr{,/**},
+  /snap/vanguard/42/usr/** rw,
+  /snap/vanguard/42/usr/ rw,
+  /snap/vanguard/42/ rw,
+  /snap/vanguard/ rw,
+  /snap/ rw,
+  /tmp/.snap/snap/vanguard/42/usr/** rw,
+  /tmp/.snap/snap/vanguard/42/usr/ rw,
+  /tmp/.snap/snap/vanguard/42/ rw,
+  /tmp/.snap/snap/vanguard/ rw,
+  /tmp/.snap/snap/ rw,
+  /tmp/.snap/ rw,
+  /tmp/ rw,
+`
+	profile2 := `  # Layout /var/cache/mylink: symlink $SNAP_DATA/link/target
+  /var/cache/mylink rwl,
```

Comment:
Why 'l'? This is for a hard link...

Reply:
Ah, my mistake - I probably added it for a symlink and it is not really needed.